core/capabilities/ccip/ccipsolana/commitcodec.go

```diff
@@ -0,0 +1,141 @@
+		token, err := solana.PublicKeyFromBase58(string(update.TokenID))
+		if err != nil {
+			return nil, fmt.Errorf("invalid token address: %s, %w", update.TokenID, err)
+		}
+		if update.Price.IsEmpty() {
+			return nil, fmt.Errorf("empty price for token: %s", update.TokenID)
+		}
+		tpu = append(tpu, ccip_router.TokenPriceUpdate{
+			SourceToken: token,
+			UsdPerToken: [28]uint8(update.Price.FillBytes(make([]byte, 28))),
+		})
+	}
+
+	gpu := make([]ccip_router.GasPriceUpdate, 0, len(report.PriceUpdates.GasPriceUpdates))
+	for _, update := range report.PriceUpdates.GasPriceUpdates {
+		if update.GasPrice.IsEmpty() {
+			return nil, fmt.Errorf("empty gas price for chain: %d", update.ChainSel)
+		}
+
+		gpu = append(gpu, ccip_router.GasPriceUpdate{
+			DestChainSelector: uint64(update.ChainSel),
+			UsdPerUnitGas:     [28]uint8(update.GasPrice.FillBytes(make([]byte, 28))),
+		})
+	}
+
+	commit := ccip_router.CommitInput{
+		MerkleRoot: mr,
+		PriceUpdates: ccip_router.PriceUpdates{
+			TokenPriceUpdates: tpu,
+			GasPriceUpdates:   gpu,
+		},
+	}
+
+	err := commit.MarshalWithEncoder(encoder)
+	if err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
+
+func (c *CommitPluginCodecV1) Decode(ctx context.Context, bytes []byte) (cciptypes.CommitPluginReport, error) {
+	decoder := agbinary.NewBorshDecoder(bytes)
+	commitReport := ccip_router.CommitInput{}
+	err := commitReport.UnmarshalWithDecoder(decoder)
+	if err != nil {
+		return cciptypes.CommitPluginReport{}, err
+	}
+
+	merkleRoots := []cciptypes.MerkleRootChain{
+		{
+			ChainSel:      cciptypes.ChainSelector(commitReport.MerkleRoot.SourceChainSelector),
+			OnRampAddress: commitReport.MerkleRoot.OnRampAddress,
+			SeqNumsRange: cciptypes.NewSeqNumRange(
+				cciptypes.SeqNum(commitReport.MerkleRoot.MinSeqNr),
+				cciptypes.SeqNum(commitReport.MerkleRoot.MaxSeqNr),
+			),
+			MerkleRoot: commitReport.MerkleRoot.MerkleRoot,
+		},
+	}
+
+	tokenPriceUpdates := make([]cciptypes.TokenPrice, 0, len(commitReport.PriceUpdates.TokenPriceUpdates))
+	for _, update := range commitReport.PriceUpdates.TokenPriceUpdates {
+		tokenPriceUpdates = append(tokenPriceUpdates, cciptypes.TokenPrice{
+			TokenID: cciptypes.UnknownEncodedAddress(update.SourceToken.String()),
+			Price:   priceHelper(update.UsdPerToken[:]),
+		})
+	}
+
+	gasPriceUpdates := make([]cciptypes.GasPriceChain, 0, len(commitReport.PriceUpdates.GasPriceUpdates))
+	for _, update := range commitReport.PriceUpdates.GasPriceUpdates {
+		gasPriceUpdates = append(gasPriceUpdates, cciptypes.GasPriceChain{
+			GasPrice: priceHelper(update.UsdPerUnitGas[:]),
+			ChainSel: cciptypes.ChainSelector(update.DestChainSelector),
+		})
+	}
+
+	return cciptypes.CommitPluginReport{
+		MerkleRoots: merkleRoots,
+		PriceUpdates: cciptypes.PriceUpdates{
+			TokenPriceUpdates: tokenPriceUpdates,
+			GasPriceUpdates:   gasPriceUpdates,
+		},
+	}, nil
+}
+
+func priceHelper(input []byte) cciptypes.BigInt {
```

Comment:
Similar note about endian-ness here, is the input in little or big endian? I believe `SetBytes` sets it in big-endian https://pkg.go.dev/math/big#Int.SetBytes

Reply:
Yeah I thought about this because Borsh encoding use little-endian. But I wonder whether this matter, as the ccip_router.TokenPriceUpdate only keeps bytes representation, and the codec decoder use big-endian when decoding.

Reply:
SVM codec should use little endian right?
This function seems to assume input is BE, while I think it should be LE

Reply:
Agreed, already converted to LE